lib/internal/process/warning.js

```diff
@@ -89,8 +94,21 @@ function doEmitWarning(warning) {
   process.emit('warning', warning);
 }
 
+let disableWarningsSet;
+
 function onWarning(warning) {
+  if (!disableWarningsSet) {
+    disableWarningsSet = new SafeSet();
+    const disableWarningsValues = getOptionValue('--disable-warnings');
+    for (let i = 0; i < disableWarningsValues.length; i++) {
+      disableWarningsSet.add(disableWarningsValues[i]);
+    }
```

Comment:
```suggestion
    disableWarningsSet = new SafeSet(getOptionValue('--disable-warnings'));
```

Reply:
This causes another test to fail. It has to do with array prototype protection. Something about Set using array iteration unsafely

Reply:
Ah okay. 👍